frame/nomination-pools/fuzzer/src/call.rs

```diff
@@ -0,0 +1,373 @@
+fn random_signed_origin<R: Rng>(rng: &mut R) -> (RuntimeOrigin, AccountId) {
+	let count = PoolMembers::<T>::count();
+	if rng.gen::<bool>() && count > 0 {
+		// take an existing account.
+		let skip = rng.gen_range(0..count as usize);
+
+		// this is tricky: the account might be our reward agent, which we never want to be
+		// randomly chosen here. Try another one, or, if it is only our agent, return a random
+		// one nonetheless.
+		let candidate = PoolMembers::<T>::iter_keys().skip(skip).take(1).next().unwrap();
+		let acc =
+			if candidate == REWARD_AGENT_ACCOUNT { rng.gen::<AccountId>() } else { candidate };
+
+		(RuntimeOrigin::signed(acc), acc)
+	} else {
+		// create a new account
+		let acc = rng.gen::<AccountId>();
+		(RuntimeOrigin::signed(acc), acc)
+	}
+}
+
+fn random_ed_multiple<R: Rng>(rng: &mut R) -> Balance {
+	let multiple = rng.gen_range(MIN_ED_MULTIPLE..MAX_ED_MULTIPLE);
+	ExistentialDeposit::get() * multiple
+}
+
+fn fund_account<R: Rng>(rng: &mut R, account: &AccountId) {
+	let target_amount = random_ed_multiple(rng);
+	if let Some(top_up) = target_amount.checked_sub(Balances::free_balance(account)) {
+		let _ = Balances::deposit_creating(account, top_up);
+	}
+	assert!(Balances::free_balance(account) >= target_amount);
+}
+
+fn random_existing_pool<R: Rng>(mut rng: &mut R) -> Option<PoolId> {
+	BondedPools::<T>::iter_keys().collect::<Vec<_>>().choose(&mut rng).map(|x| *x)
+}
+
+fn random_call<R: Rng>(mut rng: &mut R) -> (pools::Call<T>, RuntimeOrigin) {
+	let op = rng.gen::<usize>();
+	let mut op_count = <pools::Call<T> as GetCallName>::get_call_names().len();
+	// Exclude set_state, set_metadata, set_configs, update_roles and chill.
+	op_count -= 5;
+
+	match op % op_count {
+		0 => {
+			// join
+			let pool_id = random_existing_pool(&mut rng).unwrap_or_default();
+			let (origin, who) = random_signed_origin(&mut rng);
+			fund_account(&mut rng, &who);
+			let amount = random_ed_multiple(&mut rng);
+			(PoolsCall::<T>::join { amount, pool_id }, origin)
+		},
+		1 => {
+			// bond_extra
+			let (origin, who) = random_signed_origin(&mut rng);
+			let extra = if rng.gen::<bool>() {
+				BondExtra::Rewards
+			} else {
+				fund_account(&mut rng, &who);
+				let amount = random_ed_multiple(&mut rng);
+				BondExtra::FreeBalance(amount)
+			};
+			(PoolsCall::<T>::bond_extra { extra }, origin)
+		},
+		2 => {
+			// claim_payout
+			let (origin, _) = random_signed_origin(&mut rng);
+			(PoolsCall::<T>::claim_payout {}, origin)
+		},
+		3 => {
+			// unbond
+			let (origin, who) = random_signed_origin(&mut rng);
+			let amount = random_ed_multiple(&mut rng);
+			(PoolsCall::<T>::unbond { member_account: who, unbonding_points: amount }, origin)
+		},
+		4 => {
+			// pool_withdraw_unbonded
+			let pool_id = random_existing_pool(&mut rng).unwrap_or_default();
+			let (origin, _) = random_signed_origin(&mut rng);
+			(PoolsCall::<T>::pool_withdraw_unbonded { pool_id, num_slashing_spans: 0 }, origin)
```

Comment:
Would randomizing the `num_slashing_spans` make sense here and when constructing the `withdraw_unbonded` call?

Reply:
could be, although all values other than 0 would fail. 

Honestly, I am not sure about how much you should nudge a fuzzer to create random input data that's sensible (in this case `num_slashing_spans = 0`) and how much you should let it blindly try and find the right input for you.